src/workerd/api/tracing-module.h

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2017-2025 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#pragma once
+
+#include "http.h"
+#include "worker-rpc.h"
+
+#include <workerd/io/io-context.h>
+
+namespace workerd::api {
+
+// Forward declaration for InternalSpanImpl
+class InternalSpanImpl;
+
+// Wrapper class that manages span ownership through IoContext or direct ownership
+class InternalSpan: public jsg::Object {
+ private:
+  kj::OneOf<kj::Own<InternalSpanImpl>, IoOwn<InternalSpanImpl>> impl;
+
+ public:
+  InternalSpan(kj::OneOf<kj::Own<InternalSpanImpl>, IoOwn<InternalSpanImpl>> impl);
+
+  void end();
+  void setTag(jsg::Lock& js, const kj::String key, const kj::Maybe<kj::OneOf<bool, double, kj::String>> value);
```

Comment:
```suggestion
  void setTag(jsg::Lock& js, kj::String key, kj::Maybe<kj::OneOf<bool, double, kj::String>> value);
```

Also, using `kj::Maybe<...>?` here for value will require that the value be passed in as `null`. Passing in `undefined` will not be allowed. Is that what you intend? if not, then you probably want to use `jsg::Optional<....>` instead of `kj::Maybe<...>`.